src/Acuminator/Acuminator.Analyzers/Analyzers/DAC/DacPropertyGraphUsageAnalyzer.cs

```diff
@@ -0,0 +1,135 @@
+                }
+
+                IEnumerable<LocalDeclarationStatementSyntax> localVariables = node.DescendantNodes().OfType<LocalDeclarationStatementSyntax>();
+
+                foreach (LocalDeclarationStatementSyntax l in localVariables)
+                {
+                    AnalyzeLocalDeclaration(l, syntaxContext, pxContext);
+                }
+            }
+        }
+
+        private void AnalyzeLocalDeclaration(LocalDeclarationStatementSyntax local, SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            bool isGraphDeclarationType = syntaxContext.SemanticModel.GetSymbolInfo(local.Declaration.Type).Symbol is ITypeSymbol typeSymbol && typeSymbol.InheritsFromOrEquals(pxContext.PXGraphType);
+
+            if (isGraphDeclarationType)
+            {
+                syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1029_PXGraphUsageInDacProperty, local.GetLocation()));
+            }
+        }
+
+        private void AnalyzeMemberAccessExpression(ExpressionSyntax expression, SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            ISymbol expressionSymbol = syntaxContext.SemanticModel.GetSymbolInfo(expression, syntaxContext.CancellationToken).Symbol;
+
+            if (expressionSymbol != null && IsGraphUsedAsMemberExpression(expressionSymbol, pxContext))
+            {
+                syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1029_PXGraphUsageInDacProperty, expression.GetLocation()));
+            }
+        }
+
+        private void AnalyzeMemberAccessName(SimpleNameSyntax name, SyntaxNodeAnalysisContext syntaxContext, PXContext pxContext)
+        {
+            ISymbol nameSymbol = syntaxContext.SemanticModel.GetSymbolInfo(name, syntaxContext.CancellationToken).Symbol;
+
+            if (nameSymbol != null && IsGraphUsedAsMemberName(nameSymbol, pxContext))
+            {
+                syntaxContext.ReportDiagnostic(Diagnostic.Create(Descriptors.PX1029_PXGraphUsageInDacProperty, name.GetLocation()));
+            }
+        }
+
+        private bool IsGraphUsedAsMemberExpression(ISymbol expressionSymbol, PXContext pxContext)
+        {
+            bool isGraphUsed = false;
+
+            switch (expressionSymbol)
+            {
+                case ILocalSymbol l:
+                    if (l.Type.InheritsFromOrEquals(pxContext.PXGraphType))
+                    {
+                        isGraphUsed = true;
+                    }
+                    break;
+                default:
+                    isGraphUsed = IsGraphUsedAsMemberName(expressionSymbol, pxContext);
+                    break;
+            }
+
+            return isGraphUsed;
+        }
+
+        private bool IsGraphUsedAsMemberName(ISymbol nameSymbol, PXContext pxContext)
+        {
+            bool isGraphUsed = false;
+
+            switch (nameSymbol)
+            {
+                case IFieldSymbol f:
```

Comment:
@metallmonster,  some code duplication. While it's not a big deal, maybe it could be refactored together with a method IsGraphUsedAsMemberExpression to return an ITypeSymbol object to check which is checked for inheritance with PXGraph. 
Also maybe we should check for graph extensions too?